fix(Button): alignment issue

OysterD3 opened the pull request:
<!--
First of all, thank you for your contribution! 😄
For requesting to pull a new feature or bugfix, please send it from a feature/bugfix branch based on the `master` branch.
Before submitting your pull request, please make sure the checklist below is confirmed.
Your pull requests will be merged after one of the collaborators approve.
Thank you!
-->

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

fix #51811 
need to cater for #51380 as well

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |      Fix button alignment when having icon     |
| 🇨🇳 Chinese |      修复当按钮有图标时移位     |

OysterD3 committed:
fix(button): prevent extra height

OysterD3 committed:
refactor(button): update button line height

afc163 commented:
好像也不太行？去 Modal 加个 icon 看看

<img width="459" alt="图片" src="https://github.com/user-attachments/assets/a172cc96-b152-43cd-93af-78500e901a63">

OysterD3 commented:
> 好像也不太行？偏下了。去 Modal 加个 icon 看看


![image](https://github.com/user-attachments/assets/aa5ed572-04b9-40eb-bcea-83936e6b181b)

好奇怪，我这里没问题

OysterD3 commented:
看了一遍，确实有0.05px的偏差 🥲

OysterD3 committed:
fix(button): adjust icon alignment in button styles

Remove unnecessary styles for button icon display and line height. 
Change vertical alignment of SVG icons to 'middle' for better 
visual consistency across different button states.

OysterD3 committed:
style(button): adjust SVG icon alignment in button styles

Update the vertical alignment of SVG icons within buttons to 
improve visual consistency. The change specifically targets 
buttons with only an icon, ensuring better alignment with 
text elements. This adjustment enhances the overall user 
interface and adheres to design standards.

OysterD3 committed:
refactor: update icons demo

OysterD3 committed:
refactor(button): remove svg

OysterD3 commented:
@afc163 好了，但是有个问题就是用户必须确保自定义icon的大小是和button的font-size一致，要不然无解

OysterD3 committed:
test(button): update snapshots

afc163 commented on components/button/demo/debug-icon.tsx:
```diff
@@ -58,19 +51,13 @@ const App: React.FC = () => {
-            <Button icon={<Icon12Size />} />
-            <Button icon={<Icon16Size />} />
-            <Button icon={<IconIrregularSize />} />
-          </Flex>
```
别改现有的吧，可以另外加一行看效果。

DDDDD12138 replied:
我感觉是不是给 `.ant-btn ` 统一 一个 `vertical-align` 就能解决问题了呢？

OysterD3 committed:
refactor(button): revert

afc163 commented:
> 但是有个问题就是用户必须确保自定义icon的大小是和button的font-size一致，要不然无解

这个解决方案不太普适，我想想别的解。

afc163 committed:
Merge branch 'feature' into fix/button-alignment

OysterD3 committed:
refactor(button): remove redundant styles and add verticle align

Wxh16144 commented:
> [Open in Stackblitz](https://pkg.pr.new/template/548e32e3-9008-426f-b4d0-d17637a8bdd1)
> 
> More templates
> ```
> npm i https://pkg.pr.new/ant-design/ant-design/antd@51820
> ```
> 
> _commit: [`efad72e`](https://github.com/ant-design/ant-design/runs/34354637752)_


我用这个安装依赖到本地验证了一下，当前方式确实能解这几个问题


> 我的验证代码


```tsx
import React from "react";
import { Button, Modal } from "antd";
import { HighlightOutlined } from '@ant-design/icons'
import { FiColumns } from "react-icons/fi";

const App = () => {
  const slogan = `Help designers/developers building beautiful products more flexible and working with happiness`;

  return (
    <>
      <div style={{
        margin: 12,
        transformOrigin: 'top left',
        // transform: 'scale(5)'
      }}>

        {/* https://github.com/ant-design/ant-design/issues/51811 */}
        <div>
          <Button>
            OK
          </Button>
          <Button
            type="primary"
            icon={<HighlightOutlined />}
          >
            OKAbc123gf
          </Button>
        </div>

        <br />
        <br />

        {/* https://github.com/ant-design/ant-design/issues/51380 */}
        <Button
          type="primary"
          icon={<FiColumns />}
        >
          OKAbc123gf
        </Button>

        <br />
        <br />

        <div style={{ maxWidth: 500 }}>
          <Modal._InternalPanelDoNotUseOrYouWillBeFired
            okButtonProps={{
              // icon: <CloseCircleOutlined />
              icon: <FiColumns /> // https://github.com/ant-design/ant-design/issues/51380
            }}
          >
            <h4>{slogan}</h4>
          </Modal._InternalPanelDoNotUseOrYouWillBeFired>
          <hr />

          <Modal._InternalPanelDoNotUseOrYouWillBeFired
            footer={
              <>

                <Button>
                  OK
                </Button>
                <Button
                  type="primary"
                  icon={<HighlightOutlined />}
                >
                  OKAbc123gf
                </Button>
                <Button
                  icon={<FiColumns />}
                >
                  Hello
                </Button>
              </>
            }
          >
            <h4>{slogan}</h4>
          </Modal._InternalPanelDoNotUseOrYouWillBeFired>
        </div>
      </div>
    </>
  );
}

export default App;

```

> 结果

<img width="593" alt="image" src="https://github.com/user-attachments/assets/67e645ed-ed4d-4e7b-9870-a6d474c42525" />



### 补充

但是视觉回归中的变动也是对的，比如 alert 的 actions 操作。这个 PR 会导致整体往下移动一点。 感觉可以接受

<img width="1775" alt="image" src="https://github.com/user-attachments/assets/a74617f4-59ba-4d12-a584-ea477790d56f" />

afc163 commented on components/button/style/index.ts:
```diff
@@ -33,6 +33,7 @@ const genSharedButtonStyle: GenerateStyle<ButtonToken, CSSObject> = (token): CSS
       userSelect: 'none',
       touchAction: 'manipulation',
       color: token.colorText,
+      verticalAlign: 'middle',
```
会影响这个例子：https://ant.design/components/input#input-demo-align
